.travis.yml

```diff
@@ -26,117 +31,112 @@ env:
+    else
+      mkdir -p "$COZY_DESKTOP_DIR";
+    fi
 
     # Dummy .env.test (all variables are already set)
     # env-cmd ignores the file when empty
-    - echo "NODE_ENV=test" > "${TRAVIS_BUILD_DIR}/.env.test"
-
-before_script:
-    # Set up display for electron-mocha
-    - export DISPLAY=:99.0
-    - if [ "${TRAVIS_OS_NAME}" == "osx" ]; then
-        ( sudo Xvfb :99 -ac -screen 0 1024x768x8; echo ok )&
-      else
-        sh -e /etc/init.d/xvfb start;
-      fi
-    - sleep 3 # give xvfb some time to start
+    echo "NODE_ENV=test" > "${TRAVIS_BUILD_DIR}/.env.test"
 
-script:
-    - yarn build
-    - yarn lint
-    - yarn test:world --timeout $MOCHA_TIMEOUT --forbid-only
-    - yarn test:unit:coverage --timeout $MOCHA_TIMEOUT --forbid-only
-    - yarn test:elm
-    - yarn test:integration --timeout $MOCHA_TIMEOUT --forbid-only
-    - yarn test:scenarios --timeout $MOCHA_TIMEOUT --forbid-only
-    - |
-      if [ "$TRAVIS_OS_NAME" == "linux" ]; then
-        docker run --rm \
-          $(env | \
-            grep -Eo '^[^\s=]*(DEBUG|NODE_|ELECTRON_|YARN_|NPM_|CI|CIRCLE|TRAVIS|APPVEYOR_|CSC_|_TOKEN|_KEY|AWS_|STRIP|BUILD_)[^\s=]*' | \
-            sed 's/^/-e /;/^$/d' | \
-            paste -sd ' ' \
-          ) \
-          -v ${PWD}:/project \
-          -v ~/.cache/electron:/root/.cache/electron \
-          -v ~/.cache/electron-builder:/root/.cache/electron-builder \
-          electronuserland/builder:8 \
-          /bin/bash -c "yarn dist:all"
-      else
-        yarn dist:all
-      fi
+after_failure: ./script/after_failure.sh
 
-after_success:
-    - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-        bash <(curl -s https://codecov.io/bash);
-      fi
-
-after_failure:
-    - $CXX --version
-    - netstat -lntp
-    - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then cat couchdb.log; fi
+branches:
+    only:
+        - master
+stages:
+    - test
+    - name: notify
+      if: type = pull_request
+jobs:
+    fast_finish: true
+    include:
+        - &test_unit
+          stage: test
+          script: ./script/unit.sh
+          name: unit tests
+          os: linux
+        - <<: *test_unit
+          os: osx
+        - &test_integration
+          stage: test
+          script: ./script/integration.sh
+          name: integration tests
+          os: linux
+        - <<: *test_integration
+          os: osx
+        - &test_scenarios
+          stage: test
+          script: ./script/scenarios.sh
+          name: scenarios tests
+          os: linux
+        - <<: *test_scenarios
+          os: osx
+        - &dist
+          stage: test
+          script: ./script/dist.sh
+          name: artifacts build
+          os: linux
+        - <<: *dist
+          os: osx
+        - stage: test
+          name: code coverage
+          os: linux
+          before_install: skip
+          install: skip
+          after_failure: skip
+          script: bash <(curl -s https://codecov.io/bash)
```

Comment:
I think you have cheated a bit: we need to generated the coverage (`yarn test:unit:coverage`) before uploading it to codecov (`bash <(curl -s https://codecov.io/bash)`)